seed/seed16.go

```diff
@@ -147,12 +149,33 @@ func (s *seed16) addSnap(sn *internal.Snap16, pinnedTrack string, cache map[stri
 
 		var sideInfo snap.SideInfo
 		if sn.Unasserted {
+			if err := handler.HandleUnassertedSnap(sn.Name, path, tm); err != nil {
+				return nil, err
+			}
 			sideInfo.RealName = sn.Name
 		} else {
 			var si *snap.SideInfo
 			var err error
+
+			deriveRev := func(snapSHA3_384 string, snapSize uint64) (snap.Revision, error) {
+				var err error
+				if si == nil {
+					si, err = snapasserts.DeriveSideInfoFromDigestAndSize(path, snapSHA3_384, snapSize, s.db)
+				}
+				if err != nil {
+					return snap.Revision{}, err
+				}
+				return si.Revision, err
+			}
 			timings.Run(tm, "derive-side-info", fmt.Sprintf("hash and derive side info for snap %q", sn.Name), func(nested timings.Measurer) {
-				si, err = snapasserts.DeriveSideInfo(path, s.db)
+				var snapSHA3_384 string
+				var snapSize uint64
+				snapSHA3_384, snapSize, err = handler.HandleAndDigestAssertedSnap(sn.Name, path, essType, nil, deriveRev, tm)
```

Comment:
Wouldn't this work as well?
```suggestion
				snapSHA3_384, snapSize, err := handler.HandleAndDigestAssertedSnap(sn.Name, path, essType, nil, deriveRev, tm)
```

Reply:
Ah, right, thanks for the explanation!